ocaml/xapi/create_misc.ml

```diff
@@ -348,9 +348,13 @@ and update_domain_zero_record ~__context ~domain_zero_ref (host_info: host_info)
     Db.VM.set_requires_reboot ~__context ~self:domain_zero_ref ~value:false
   end;
   let localhost = Helpers.get_localhost ~__context in
+  let metrics   = Db.VM.get_metrics ~__context ~self:domain_zero_ref in
   Db.VM.set_power_state ~__context ~self:domain_zero_ref ~value:`Running;
   Db.VM.set_domid ~__context ~self:domain_zero_ref ~value:0L;
-  Helpers.update_domain_zero_name ~__context localhost host_info.hostname
+  Helpers.update_domain_zero_name ~__context localhost host_info.hostname;
+  Db.VM_metrics.set_VCPUs_number ~__context ~self:metrics
```

Comment:
You also need to update `VM.VCPUs_max`, `VM.VCPUs_at_startup` and `VM.VCPUs_params`. See where all `vcpus` is used in `create_domain_zero_record`.

Reply:
`VM.VCPUs_params` is a list. But what about `vCPUs_utilisation`?

Reply:
Oh yes, not VCPU_params. But yes to `VCPU_utilisation` in the metrics.

Reply:
Cleaner would be to update the metrics from `ensure_domain_zero_metrics_record` (perhaps add `update_domain_zero_metrics_record`, since all those function are split between the main record and the metrics.

Reply:
I've created `update_domain_zero_metrics_record` but I don't see how `vCPUs_utilisation` could be updated.

Reply:
I believe this has been addressed.